game/song.cc

```diff
@@ -15,7 +15,7 @@ extern "C" {
 }
 
 template <typename T> std::optional<T> getJsonEntry(nlohmann::json const& json, const char *name) {
```

Comment:
Template should live in json.hh/cc

Reply:
Agreed, and please also keep the count() > 0 (it enable you to remove the check in FindJson.cmake

Reply:
The template still lives in game/song.cc please move it over to game/json.hh